services-geojson/src/main/java/com/mapbox/geojson/BaseCoordinatesTypeAdapter.java

```diff
@@ -25,8 +25,11 @@
 abstract class BaseCoordinatesTypeAdapter<T> extends TypeAdapter<T> {
 
 
-  protected void writePoint(JsonWriter out, Point value) throws  IOException {
-    writePointList(out, value.coordinates());
+  protected void writePoint(JsonWriter out, Point point) throws  IOException {
+    if (point == null) {
```

Comment:
We are now choosing to ignore nulls, while we should actually be able to support them when serializing and deserializing `List`s in the `BaseCoordinatesTypeAdapter`. I think we'll need to refactor both read and write methods.

Reply:
@LukasPaczos what logic should `writePoint` has in case of `null` Point?
also `readPoint` always returns `non null` Point or throws an exception.

Reply:
> also readPoint always returns non null Point or throws an exception

That's what we'd need to refactor. Could we use `JsonToken.NULL` or a similar construct?

Reply:
Is this because before `null` wasn't an option when reach serialization / deserialization and now it is? I believe the problem is that before we were always dealing with `String`s and not anymore 🤔 How serialization / deserialization was working before?

Reply:
@Guardiola31337 @LukasPaczos  I think it's only because of `waypointTargetsList()` in `RouteOptions`. 
Previously waypointTargets were stored as a String, so `PointAsCoordinatesTypeAdapter` was not used. But now they are a `List<Point>`, so `PointAsCoordinatesTypeAdapter` calls `BaseCoordinatesTypeAdapter.writePoint()`.
It is possible to store `nulls` in `waypointTargetsList`, so we need to check for a null inside `writePoint()` method

Reply:
@korshaknn could we add some tests in `RouteOptions` verifying that the serialization / deserialization works 👌 (`toJson` / `fromJson`)?

Reply:
That's right @korshaknn, but we still need to serialize and deserialize this `null`, not skip it. I think the inconsistency will be caught with a test that @Guardiola31337 suggested.

Reply:
@LukasPaczos how that `null` Point should be serialized?

Reply:
> how that `null` Point should be serialized?

Probably `""` (empty)? Could we run a quick test of `toJson` / `fromJson` before and after https://github.com/mapbox/mapbox-java/pull/1118 to check how was working, compare and do the same?

Reply:
Empty string sounds correct.